api/src/main/java/org/openmrs/module/sync2/api/sync/SyncClient.java

```diff
@@ -130,38 +131,54 @@ private void prepareRestTemplate(ClientHelper clientHelper) {
 		restTemplate.setMessageConverters(converters);
 	}
 
-	private Object retrieveObject(String category, String resourceUrl, String destinationUrl, String clientName)
+	private Object retrieveObject(String category, String resourceUrl, String destinationUrl, String clientName,
+			OpenMRSSyncInstance instance)
 			throws RestClientException, URISyntaxException {
 		ClientHelper helper = ClientHelperFactory.createClient(clientName);
 		Class<?> clazz = helper.resolveClassByCategory(category);
 
-		return restTemplate.exchange(sendRequest(category, destinationUrl, clientName,
-				new InnerRequest(helper.retrieveRequest(resourceUrl))),
-				clazz).getBody();
+		RequestEntity request = helper.retrieveRequest(resourceUrl);
+		if (!SyncUtils.clientHasSpecificAddress(clientName, instance)) {
+			request = sendRequest(category, destinationUrl, clientName, new InnerRequest(request));
+		}
+		return exchange(request, clazz).getBody();
```

Comment:
are you switching between wrapped and unwrapped request?

Reply:
Maybe rename clientHasSpecificAddress method or move this if statement to separate method

Reply:
ok